KeyboardAvoidanceSwiftUI/KeyboardAdaptive.swift

```diff
@@ -12,24 +12,38 @@ import Combine
 /// Note that the `KeyboardAdaptive` modifier wraps your view in a `GeometryReader`, 
 /// which attempts to fill all the available space, potentially increasing content view size.
 struct KeyboardAdaptive: ViewModifier {
+    private let rerender: AnyPublisher<Any, Never>
     @State private var bottomPadding: CGFloat = 0
-
+    
+    init(rerender: AnyPublisher<Any, Never>? = nil) {
+        var publishers = [Publishers.keyboardHeight.map { arg -> Any in arg }.eraseToAnyPublisher()]
+        if let rerender = rerender {
+            publishers.append(rerender)
+        }
+        self.rerender = Publishers.MergeMany(publishers).eraseToAnyPublisher()
+    }
+    
+    private func bottomPadding(forGeometry geometry: GeometryProxy) -> CGFloat {
+        let keyboardTop = geometry.frame(in: .global).height - Publishers.keyboardHeight.value
+        var focusedTextInputBottom = UIResponder.currentFirstResponder?.globalFrame?.maxY ?? 0
+        focusedTextInputBottom += bottomPadding / 2
```

Comment:
I'm dividing by 2 and multiplying by 2 here for reasons described [here](https://github.com/V8tr/KeyboardAvoidanceSwiftUI/issues/1#issuecomment-634335177). My keyboard is around 500pt tall, like I mention in the comments there.